pkg/services/oauth/oauth.go

```diff
@@ -56,34 +95,196 @@ func (s *service) Configure() (err error) {
+	// checks signing certificate and sets vendor, actorName in validationContext
+	if err := s.validateIssuer(&context); err != nil {
+		return nil, err
+	}
+
+	// check if client certificate is issued by vendor according to RFC003 §5.2.1.2
+	if err := s.validateClientCertificate(&context, request.ClientCert); err != nil {
+		return nil, err
+	}
+
+	// check if the custodian is registered by this vendor, according to RFC003 §5.2.1.8
+	if err := s.validateSubject(&context); err != nil {
+		return nil, err
+	}
+
+	// Validate the AuthTokenContainer, according to RFC003 §5.2.1.5
+	// todo: request.VendorIdentifier is deprecated, remove in 0.17
+	var err error
+	if context.contractValidationResult, err = s.contractValidator.ValidateJwt(context.jwtBearerToken.AuthTokenContainer, request.VendorIdentifier); err != nil {
 		return nil, fmt.Errorf("identity token validation failed: %w", err)
 	}
-	if res.ValidationResult == services.Invalid {
+	if context.contractValidationResult.ValidationResult == services.Invalid {
 		return nil, fmt.Errorf("identity validation failed")
 	}
+	// checks if the name from the login contract matches with the registered name of the issuer.
+	if err = s.validateActor(&context); err != nil {
+		return nil, err
+	}
+
+	// validate the endpoint in aud, according to RFC003 §5.2.1.6
+	// the aud field must have the identifier of the endpoint registered by the vendor of this node!
+	// this is needed to prevent relay attacks.
+	// todo: implement when services and endpoints in registry have been implemented (https://github.com/nuts-foundation/nuts-registry/issues/156)
+
+	// validate the legal base, according to RFC003 §5.2.1.7 if sid is present
+	if err = s.validateLegalBase(context.jwtBearerToken); err != nil {
+		return nil, err
+	}
 
-	accessToken, err := s.buildAccessToken(jwtBearerToken, res)
+	accessToken, err := s.buildAccessToken(&context)
 	if err != nil {
 		return nil, err
 	}
 
 	return &services.AccessTokenResult{AccessToken: accessToken}, nil
 }
 
+// checks if the name from the login contract matches with the registered name of the issuer.
+func (s *service) validateActor(context *validationContext) error {
+	if context.contractValidationResult.ContractAttributes[contract.LegalEntityAttr] != context.actorName {
+		return errors.New("legal entity mismatch")
+	}
+	return nil
+}
+
+// check the actor against the registry, according to RFC003 §5.2.1.3
+// we do this by getting the validation chain for the certificate in the x5c header and check the vendorID SAN from the root
+// with the vendorId of the actor. It returns the actor name which must match the login contract.
+func (s *service) validateIssuer(context *validationContext) error {
+	validationTime := time.Unix(context.jwtBearerToken.IssuedAt, 0)
+	var vendor core.PartyID
+
+	actorPartyID, err := core.ParsePartyID(context.jwtBearerToken.Issuer)
+	if err != nil {
+		return fmt.Errorf(errInvalidIssuerFmt, err)
```

Comment:
if we're not including the invalid partyID, do we have enough logging to analyze PRD issues?

(same for other errs below here)

Reply:
added the erroneous string